src/main/java/picard/vcf/AccumulateVariantCallingMetrics.java

```diff
@@ -0,0 +1,132 @@
+
+    @Override
+    protected int doWork() {
+
+        // set up the collectors
+        final Map<String, List<CollectVariantCallingMetrics.VariantCallingDetailMetrics>> sampleDetailsMap = new HashMap<>();
+        final Collection<CollectVariantCallingMetrics.VariantCallingSummaryMetrics> summaries = new ArrayList<>();
+
+        for (final File file : INPUT) {
+            final String inputPrefix = file.getAbsolutePath() + ".";
+
+            try {
+                // read in the detailed metrics file
+                final File detail = new File(inputPrefix + CollectVariantCallingMetrics.VariantCallingDetailMetrics.getFileExtension());
+                IOUtil.assertFileIsReadable(detail);
+                MetricsFile<CollectVariantCallingMetrics.VariantCallingDetailMetrics, ?> detailedMetricsFile = getMetricsFile();
+                detailedMetricsFile.read(new FileReader(detail));
+
+                // for each sample in the detailed metrics...
+                long totalHetDepth = 0L;
+                for (final CollectVariantCallingMetrics.VariantCallingDetailMetrics detailedMetrics : detailedMetricsFile.getMetrics()) {
+                    // re-calculate internal fields from derived fields
+                    detailedMetrics.calculateFromDerivedFields();
+                    totalHetDepth += detailedMetrics.TOTAL_HET_DEPTH;
+
+                    // add it to the list of metrics for that sample so that we can merge them later
+                    if (!sampleDetailsMap.containsKey(detailedMetrics.SAMPLE_ALIAS)) {
+                        sampleDetailsMap.put(detailedMetrics.SAMPLE_ALIAS, new ArrayList<>());
+                    }
+                    sampleDetailsMap.get(detailedMetrics.SAMPLE_ALIAS).add(detailedMetrics);
+                }
+
+                // next, read in the summary metrics
+                final File summary = new File(inputPrefix + CollectVariantCallingMetrics.VariantCallingSummaryMetrics.getFileExtension());
+                IOUtil.assertFileIsReadable(summary);
+                MetricsFile<CollectVariantCallingMetrics.VariantCallingSummaryMetrics, ?> summaryMetricsFile = getMetricsFile();
+                summaryMetricsFile.read(new FileReader(summary));
+
+                // re-calculate internal fields from derived fields and add it to the list of summary metrics
+                final CollectVariantCallingMetrics.VariantCallingSummaryMetrics summaryMetrics = summaryMetricsFile.getMetrics().get(0);
+                summaryMetrics.calculateFromDerivedFields(totalHetDepth);
+                summaries.add(summaryMetrics);
+            } catch (IOException e) {
+                throw new PicardException(String.format("Cannot read from metrics files with prefix %s", inputPrefix));
+            }
+        }
+
+        // now merge all of the accumulated metrics
+        final Collection<CollectVariantCallingMetrics.VariantCallingDetailMetrics> collapsedDetails = new ArrayList<>();
```

Comment:
I rewrote the code to address this and didn't like the new version.  You either need to do extra compute (to recalculate derived fields) so many unneeded times or you leave the summary collection in a dangerously inconsistent state.  I'd like to advocate that we keep the code as is since it's safer.